sql/core/src/test/resources/sql-tests/inputs/transform.sql

```diff
@@ -121,6 +121,38 @@ USING 'cat' AS (d)
   NULL DEFINED AS 'NULL'
 FROM t;
 
+SELECT TRANSFORM(a, b, c, null)
+  ROW FORMAT DELIMITED
+  FIELDS TERMINATED BY '@'
+  LINES TERMINATED BY '\n'
+USING 'cat' AS (a, b, c, d)
+  ROW FORMAT DELIMITED
+  FIELDS TERMINATED BY '@'
+  LINES TERMINATED BY '\n'
+  NULL DEFINED AS 'NULL'
```

Comment:
What if `NULL DEFINED` includes a meta character, e.g., `\n`? It works well?

Reply:
> What if `NULL DEFINED` includes a meta character, e.g., `\n`? It works well?

No, also in hive
![image](https://user-images.githubusercontent.com/46485123/125881169-4cbd510c-0a61-4125-8045-94073764f150.png)

Reply:
Can you explain it instead of just throwing out a screenshot? What's the behavior of Hive if `NULL DEFINED AS '\n'`?

Reply:
ping @AngersZhuuuu

Reply:
> Can you explain it instead of just throwing out a screenshot? What's the behavior of Hive if `NULL DEFINED AS '\n'`?

Sorry for miss this.

If hive set `NULL DEFINED AS '\n'`, null value will be converted to `\n` and when handle output, we split  line with `\n` first.  then the string `\n` for replacing null value was regarded as line separator too and split a new line.
Such as we have one line inpurt
```
1@true@Spark SQL@\n
```

then it will split to two line output
```
1@true@Spark SQL@

```
the first line will be split as "1", "true", "Spark SQL", ""
the second line was split as ""
After pad null value.the result is 
```
1   true        Spark SQL
     NULL     NULL              NULL
```